library/ziti_src.c

```diff
@@ -74,16 +74,40 @@ int ziti_src_init(uv_loop_t *l, um_src_t *st, const char *svc, ziti_context ztx)
     uv_link_init(st->link, &ziti_link_methods);
 
     ziti_link_t *zl = (ziti_link_t *) st->link;
-    zl->service = strdup(svc);
+    if (svc)
+        zl->service = strdup(svc);
+    else
+        zl->service = NULL;
     zl->ztx = ztx;
     
     return 0; 
 }
 
 static int
-ziti_src_connect(um_src_t *src, const char *h, const char *p, um_src_connect_cb cb, void *conn_ctx) {
+ziti_src_connect(um_src_t *src, const char *host, const char *port, um_src_connect_cb cb, void *conn_ctx) {
     ziti_link_t *zl = (ziti_link_t *) src->link;
 
+    char app_data[1024];
+    size_t app_data_len = snprintf(app_data, sizeof(app_data), "{"
+                                         "\"dst_protocol\":\"tcp\","
+                                         "\"dst_hostname\":\"%s\", "
+                                         "\"dst_port\":\"%s\""
+                                         "}", host, port);
+    ziti_dial_opts opts = {
+            .app_data = app_data,
+            .app_data_sz = app_data_len,
```

Comment:
we should probably check that `snprintf` returned a positive value before getting here.

Reply:
`snprintf` always returns a positive value

Reply:
ahh, TIL C99 compliant implementations don't return -1 when the buffer is too small.